src/soundio/soundmanager.cpp

```diff
@@ -623,20 +623,14 @@ void SoundManager::readProcess() const {
 }
 
 void SoundManager::registerOutput(const AudioOutput& output, AudioSource* src) {
-    if (m_registeredSources.contains(output)) {
-        qDebug() << "WARNING: AudioOutput already registered!";
+    VERIFY_OR_DEBUG_ASSERT(!m_registeredSources.contains(output)) {
+        return;
     }
     m_registeredSources.insert(output, src);
     emit outputRegistered(output, src);
 }
 
 void SoundManager::registerInput(const AudioInput& input, AudioDestination* dest) {
-    if (m_registeredDestinations.contains(input)) {
```

Comment:
We should leave a comment why registering an input twice is ok. Otherwise it might be confusing why outputs are handled differently with the debug assertion. Inserting the text from the PR description should be sufficient to preserve this knowledge.

Reply:
done